src/System.Net.Http/src/System/Net/Http/Headers/ObjectCollection.cs

```diff
@@ -23,6 +24,13 @@ public ObjectCollection(Action<T> validator)
             _validator = validator;
         }
 
+        // This is only used internally to enumerate the collection
+        // without the enumerator allocation.
+        new public List<T>.Enumerator GetEnumerator()
```

Comment:
We should probably consider adding a strongly-typed, struct-based enumerator to ```Collection<T>```; I was a bit surprised it didn't already have one.